lib/ansible/modules/cloud/amazon/rds_option_group.py

```diff
@@ -0,0 +1,384 @@
+
+    try:
+        response = client.create_option_group(**params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    results = response
+    return changed,results
+
+
+def match_option_group_options(client, module):
+    requires_update = False
+    new_options = module.params.get('options')
+
+    # Get existing option groups and compare to our new options spec
+    current_option_group = get_option_group(client, module)
+    for current_option in current_option_group['OptionGroupsList']:
+        if current_option['Options'] == []:
+            requires_update = True
+        else:
+            for option in current_option['Options']:
+                for setting_name in new_options:
+                    if setting_name['OptionName'] == option['OptionName']:
+                        for name in setting_name.iterkeys():
+                            # Security groups need to be handled separately due to different keys on request and what is
+                            # returned by the API
+                            if name in option.iterkeys() and name != 'OptionSettings' and name != 'VpcSecurityGroupMemberships':
+                                if cmp(setting_name[name], option[name]) != 0:
+                                    requires_update = True
+                            if name in option.iterkeys() and name == 'VpcSecurityGroupMemberships':
+                                for groups in option[name]:
+                                    if groups['VpcSecurityGroupId'] not in setting_name[name]:
+                                        requires_update = True
+
+                    for new_option in new_options:
+                        if option['OptionName'] == new_option['OptionName']:
+                            for current_option_setting in option['OptionSettings']:
+                                for new_option_setting in new_option['OptionSettings']:
+                                    if new_option_setting['Name'] == current_option_setting['Name']:
+                                        if cmp(new_option_setting['Value'], current_option_setting['Value']) != 0:
+                                            requires_update = True
+
+    return requires_update
+
+
+def compare_option_group(client, module):
+    current_option_group = get_option_group(client, module)
+    new_options = module.params.get('options')
+    to_be_added = None
+    to_be_removed = None
+    for current_option in current_option_group['OptionGroupsList']:
+        # Catch new settings you've provided that aren't in the current settings
+        old_settings = []
+        new_settings = []
+        for setting_name in new_options:
+            new_settings.append(setting_name['OptionName'])
+        for option in current_option['Options']:
+            old_settings.append(option['OptionName'])
+        if cmp(sorted(old_settings),sorted(new_settings)) != 0:
```

Comment:
This whole section would be much simpler if you started with `set`s and continued with them

```
new_settings = set([item['OptionName'] for item in new_options])
old_settings = set([item['OptionName'] for item in current_option['OptionName'])
if new_settings != old_settings:
    to_be_added = list(new_settings - old_settings)
    to_be_removed = list(old_settings - new_settings)
```